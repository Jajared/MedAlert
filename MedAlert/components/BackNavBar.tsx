import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

export default function BackNavBar({ title, navigation }) {
  return (
    <View style={styles.container} testID="BackNavBar">
      <TouchableOpacity onPress={() => navigation.goBack()} style={styles.backButton} testID="BackButton">
        <Ionicons name="arrow-back" size={24} color="black" />
      </TouchableOpacity>
      <Text style={styles.title}>{title}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#fff",
    height: 50,
    paddingHorizontal: 16,
    width: "100%",
    justifyContent: "center",
  },
  backButton: {
    marginLeft: -20,
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
    marginLeft: 16,
  },
});
