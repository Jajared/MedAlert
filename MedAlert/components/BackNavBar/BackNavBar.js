import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

export default function BackNavBar({ title, props, navigation }) {
  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={() => navigation.goBack()} style={styles.backButton}>
        <Ionicons name="arrow-back" size={24} color="black" />
      </TouchableOpacity>
      <Text style={styles.title}>{title}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#fff",
    height: 50,
    paddingHorizontal: 16,
    width: "90%",
    justifyContent: "center",
  },
  backButton: {
    marginLeft: -20,
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
    marginLeft: 16,
  },
});
