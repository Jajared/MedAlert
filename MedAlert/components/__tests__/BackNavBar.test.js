import React from "react";
import renderer from "react-test-renderer";

import BackNavBar from "../BackNavBar";

describe("<BackNavBar />", () => {
  it("has 2 child", () => {
    const tree = renderer.create(<BackNavBar />).toJSON();
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<BackNavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
