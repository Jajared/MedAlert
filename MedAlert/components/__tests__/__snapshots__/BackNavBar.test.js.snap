// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackNavBar /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "paddingHorizontal": 16,
      "width": "90%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginLeft": -20,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "black",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ï„¤
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginLeft": 16,
      }
    }
  />
</View>
`;
